pallets/prices/src/lib.rs

```diff
@@ -154,6 +160,19 @@ pub mod pallet {
             <Pallet<T> as EmergencyPriceFeeder<CurrencyId, Price>>::reset_emergency_price(asset_id);
             Ok(().into())
         }
+
+        /// Set foreign vault token mapping
+        #[pallet::weight((<T as Config>::WeightInfo::set_foreign_vault_mapping(), DispatchClass::Operational))]
+        #[transactional]
+        pub fn set_foreign_vault_mapping(
```

Comment:
if we are doing  mapping from assetId to assetId then it can be more general and has nothing to do with `vault` anymore.

if we are using `vault` wording, I think then we can map from assetId to `leaseStart, leaseEnd`